packages/mask/src/plugins/Trader/trader/0x/useTradeCallback.ts

```diff
@@ -40,9 +40,26 @@ export function useTradeCallback(tradeComputed: TradeComputed<SwapQuoteResponse>
             type: TransactionStateType.WAIT_FOR_CONFIRMING,
         })
 
+        const config_ = {
+            ...config,
+            gas: await web3.eth
+                .estimateGas({
+                    from: account,
+                    ...pick(tradeComputed.trade_, ['to', 'data', 'value']),
+                })
+                .catch((error) => {
+                    console.log(error)
+                    setTradeState({
+                        type: TransactionStateType.FAILED,
+                        error,
+                    })
+                    return 0
+                }),
+        }
+
         // estimate transaction
         try {
```

Comment:
The reason didn't do estimation for ZRX is sometimes it reports failure on estimation but succeeds on the chain. IDK the internal reason since only did a `call` here.

Reply:
Ok, i will remove it.